src/foam/dao/NoSelectDAO.js

```diff
@@ -0,0 +1,38 @@
+foam.CLASS({
+  package: 'foam.dao',
+  name: 'NoSelectDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  requires: [
+    'foam.dao.ArraySink',
+    'foam.mlang.predicate.Predicate',
+    'foam.mlang.Expr'
+  ],
+
+  methods: [
+    {
+      name: 'select_',
+      code: async function select_ () {
+        return this.ArraySink.create();
```

Comment:
Won't this cause problems for when client code passes in some other type of DAO and then cast it back to their Sink type? Why not just return the passed in sink? What are you trying to do actually?